source/Octopus.Client/Util/ParameterHelper.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Octopus.Client.Exceptions;
+using Octopus.Client.Repositories.Async;
+
+namespace Octopus.Client.Util
+{
+    static class ParameterHelper
+    {
+        public static Dictionary<string, object> CombineParameters(Dictionary<string, object> currentAdditionalQueryParameters, object userProvidedParameters)
+        {
+            if (currentAdditionalQueryParameters == null)
```

Comment:
You can make the default implementation in `BasicRepository` return an empty dictionary to avoid this null check